src/Tools/ExternalAccess/LiveShare/CodeActions/RoslynCodeActionProvider.cs

```diff
@@ -53,31 +54,39 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 Range = ProtocolConversions.TextSpanToRange(span, text)
             };
 
-            var commands = await lspClient.RequestAsync(LSP.Methods.TextDocumentCodeAction, codeActionParams, cancellationToken).ConfigureAwait(false);
+            var commands = await lspClient.RequestAsync(LSP.Methods.TextDocumentCodeAction.ToLSRequest(), codeActionParams, cancellationToken).ConfigureAwait(false);
             if (commands == null)
             {
                 return;
             }
 
             foreach (var command in commands)
             {
-                if (command is LSP.Command lspCommand)
+                LSP.Command lspCommand = null;
+                try
+                {
+                    lspCommand = ((JObject)command).ToObject<LSP.Command>();
+                }
+                catch (Exception) { }
+                if (lspCommand != null)
                 {
                     // The command can either wrap a Command or a CodeAction.
                     // If a Command, leave it unchanged; we want to dispatch it to the host to execute.
                     // If a CodeAction, unwrap the CodeAction so the guest can run it locally.
                     var commandArguments = lspCommand.Arguments.Single();
 
+                    LSP.CodeAction lspCodeAction = null;
+                    try
+                    {
+                        lspCodeAction = ((JObject)commandArguments).ToObject<LSP.CodeAction>();
+                    }
+                    catch (Exception) { }
```

Comment:
I also don't like this one - we're checking if it's an LSP code action by attempting to de-serialize it and catching the exceptions.  I looked a bit and there doesn't seem to be a great way to check.

Maybe extract out a method/static local function
```csharp
bool TryGetCodeAction(object arguments, out LSP.CodeAction codeAction)
{
... do the try catch here
}
```
That'll at least help understand what it is trying to do.

Reply:
Sure, we could do that. And yes, there's no better way to validate that the JSON you're trying to deserialize is actually of a specific type without doing the try/catch (or if you have JSON schema).

Reply:
What do you think of my implementation?

Reply:
Nice!  Much easier to read and see what's going on.